tests/test_core/test_annotationdb.py

```diff
@@ -0,0 +1,633 @@
+from cogent3 import load_unaligned_seqs, open_
+from cogent3.core.annotation_db import GenbankAnnotationDb, GffAnnotationDb
+from cogent3.parse.genbank import MinimalGenbankParser
+from cogent3.parse.gff import gff_parser
+import os
+
+base_path = os.path.dirname(os.path.dirname(__file__))
+data_path = os.path.join(base_path, "data")
+gff_path = os.path.join(data_path, "prok_NoLocusTags.gff")
+genbank_path = os.path.join(data_path, "NC_000913.3.gb")
+fasta_path = os.path.join(data_path, "short.fa")
+
+
+__author__ = "Kirat Alreja, Gavin Huttley"
+__copyright__ = "Copyright 2007-2022, The Cogent Project"
+__credits__ = ["Rob Knight", "Peter Maxwell", "Matthew Wakefield", "Gavin Huttley"]
+__license__ = "BSD-3"
+__version__ = "2022.8.24a1"
+__maintainer__ = "Gavin Huttley"
+__email__ = "Gavin.Huttley@anu.edu.au"
+__status__ = "Prototype"
+
+
+def test_make_db():
+    """Test if the gff3 database is correctly created
+    in terms of the names of columns & the number of entries"""
+
+    db = GffAnnotationDb(gff_path)
+    sql_columns = []
+    data = db.db.execute("""SELECT * FROM GFF""")
+    for column in data.description:
+        sql_columns.append(column[0])
+    got = sql_columns
+    parsed_gff = gff_parser(gff_path)
+    expected = list((list(parsed_gff)[0]).keys())[:-1]
+    assert got == expected
+
+
+def test_make_sql_query():
+    """Test that the SQLlite queries are correctly formed"""
+
+    db = GffAnnotationDb(gff_path)
+
+    """only the bio_type is provided"""
+    expected = ("SELECT * FROM GFF WHERE Type == ?", ["gene"])
+    got = db._make_sql_query(
+        seq_name=None, bio_type="gene", identifier=None, start=None, end=None
+    )
+    assert got == expected
+
+    """only the identifier is provided"""
+    expected = ("SELECT * FROM GFF WHERE Attributes like ?", ["%RandomIdentifier%"])
+    got = db._make_sql_query(
+        seq_name=None,
+        bio_type=None,
+        identifier="RandomIdentifier",
+        start=None,
+        end=None,
+    )
+    assert got == expected
+
+    """both identifier and bio_type provided"""
```

Comment:
remove all literal strings within functions, you can leave the docstrings under the function definition.